lib/autoupdate/agent.go

```diff
@@ -20,17 +20,38 @@ package autoupdate
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"errors"
 	"io/fs"
 	"log/slog"
+	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
+	"time"
 
 	"github.com/google/renameio/v2"
 	"github.com/gravitational/trace"
 	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/api/client/webclient"
+	libdefaults "github.com/gravitational/teleport/lib/defaults"
+	libutils "github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	// cdnURITemplate is the default template for the Teleport tgz download.
+	cdnURITemplate = "https://cdn.teleport.dev/teleport-v{{.Version}}-{{.OS}}-{{.Arch}}-bin.tar.gz"
```

Comment:
should we download teleport package if current agent built with enterprise or fips?

Reply:
Thanks, will add